monkey.py

```diff
@@ -11,7 +11,7 @@
     from _typeshed import StrPath
     from types import ModuleType
 
-best_name = functools.cache(discovery.best_name)
+best_name = functools.lru_cache(maxsize=None)(discovery.best_name)
```

Comment:
I think I'd like to wrap this in a compat module to signal that cache can be used starting with Python 3.9, but I'll do that subsequently.

Reply:
```suggestion
from jaraco.compat.py38 import cache
best_name = cache(discovery.best_name)
```